### ISSUE TITLE
Bug in list creation when using Chinese parameter for AddItem

### ISSUE THREAD

**Issue by nohunt** [rivo/tview#862]
OS: Windows 11 

Description:
I modified the list demo in the demos directory by changing the parameter of AddItem from English to Chinese. Here's the modified code:

```
package main

import (
	"github.com/rivo/tview"
)

func main() {
	app := tview.NewApplication()
	list := tview.NewList().
		AddItem("列表 1", "Some explanatory text", 'a', nil).
		AddItem("List item 2", "Some explanatory text", 'b', nil).
		AddItem("List item 3", "Some explanatory text", 'c', nil).
		AddItem("List item 4", "Some explanatory text", 'd', nil).
		AddItem("Quit", "Press to exit", 'q', func() {
			app.Stop()
		})

	if err := app.SetRoot(list, true).EnableMouse(true).Run(); err != nil {
		panic(err)
	}
}
```

After compiling and running the code, the following issue occurs: when selecting a Chinese item and pressing the Enter key, the content disappears, and the arrow keys also don't work properly. The same behavior happens when using the up and down arrow keys.

![1](https://github.com/rivo/tview/assets/19381274/b0bf019d-38fb-41c0-8644-f011ac6bd844)
![2](https://github.com/rivo/tview/assets/19381274/5aa0f713-3ac6-447b-8f1d-6c4943cf6b1b)

**Comment by nohunt** [comment]
> OS: Windows 11
> 
> Description: I modified the list demo in the demos directory by changing the parameter of AddItem from English to Chinese. Here's the modified code:
> 
> ```
> package main
> 
> import (
> 	"github.com/rivo/tview"
> )
> 
> func main() {
> 	app := tview.NewApplication()
> 	list := tview.NewList().
> 		AddItem("列表 1", "Some explanatory text", 'a', nil).
> 		AddItem("List item 2", "Some explanatory text", 'b', nil).
> 		AddItem("List item 3", "Some explanatory text", 'c', nil).
> 		AddItem("List item 4", "Some explanatory text", 'd', nil).
> 		AddItem("Quit", "Press to exit", 'q', func() {
> 			app.Stop()
> 		})
> 
> 	if err := app.SetRoot(list, true).EnableMouse(true).Run(); err != nil {
> 		panic(err)
> 	}
> }
> ```
> 
> After compiling and running the code, the following issue occurs: when selecting a Chinese item and pressing the Enter key, the content disappears, and the arrow keys also don't work properly. The same behavior happens when using the up and down arrow keys.
> 
> ![1](https://user-images.githubusercontent.com/19381274/251663293-b0bf019d-38fb-41c0-8644-f011ac6bd844.jpg) ![2](https://user-images.githubusercontent.com/19381274/251663318-5aa0f713-3ac6-447b-8f1d-6c4943cf6b1b.jpg)

Apologies, I found a solution to the issue in someone else's reported problem. I added the following code snippet, as mentioned below. However, I'm unsure about the underlying reason for its effectiveness

```
list.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
		app.Sync()
		return event
	})
```

**Comment by rivo** [comment]
Isn't this an issue with the Windows terminal? I thought that it cannot display Unicode characters properly. On macOS iTerm 2, it looks like this:

<img width="209" alt="image" src="https://github.com/rivo/tview/assets/480930/f4184463-8c04-4617-8c11-cdcfdfcfc20a">

**Comment by nohunt** [comment]
> Isn't this an issue with the Windows terminal? I thought that it cannot display Unicode characters properly. On macOS iTerm 2, it looks like this:

I'm not entirely sure if it's an issue with Windows terminal, but after modifying the code, it appears to have resolved the problem. Here's the updated code

```
package main

import (
	"github.com/gdamore/tcell/v2"
	"github.com/rivo/tview"
)

func main() {
	app := tview.NewApplication()
	list := tview.NewList().
		AddItem("列表 1", "Some explanatory text", 'a', nil).
		AddItem("List item 2", "Some explanatory text", 'b', nil).
		AddItem("List item 3", "Some explanatory text", 'c', nil).
		AddItem("List item 4", "Some explanatory text", 'd', nil).
		AddItem("Quit", "Press to exit", 'q', func() {
			app.Stop()
		})

	list.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
		app.Sync()
		return event
	})
	if err := app.SetRoot(list, true).EnableMouse(true).Run(); err != nil {
		panic(err)
	}
}
```
![Dingtalk_20230710104119](https://github.com/rivo/tview/assets/19381274/dd118ca8-9376-4161-8c6e-cb4ff737f1a8)

**Comment by unitwk** [comment]
Great, it works.

I also encountered the same problem.

Although I think this should be a `BUG`, but at least there is a solution.

**Comment by xyaman** [comment]
I have exactly the same problem when using wezterm on Windows 11. Using App.Sync() works, but with every event, it blinks, so it's not ideal.

**Comment by rivo** [comment]
I ran the code from above ([the original comment, *without* the `app.Sync()` command](https://github.com/rivo/tview/issues/862#issue-1792934549)) on Windows 11 with no problems. No flickering, no disappearing characters after pressing <kbd>Enter</kbd> or otherwise. Here is a screenshot:

![image](https://github.com/rivo/tview/assets/480930/a8e302ee-1fa3-4c65-980d-f64e9beb48e2)

The flickering on Windows has been resolved a few weeks ago. (Not sure if this was the original problem but I cannot reproduce it as of now.)

I don't have wezterm so I don't know how compatible it is with `tcell`. It seems to me that this is rather a problem with your terminal's configuration.

In general, I should mention that `tview` does not deal with terminal configurations or escape characters directly. This all happens on the `tcell` layer. I would suggest that you create a small program using `tcell` (*not* using any `tview` components, to avoid confusion) which reproduces this and then open an issue there: https://github.com/gdamore/tcell.

So I will close this issue now as I don't see anything that can be solved in this project.
